packages/cli/test/functional/testUtil/compare.js

```diff
@@ -24,26 +25,43 @@ function _readFileSync(...paths) {
   return fs.readFileSync(path.resolve(...paths), 'utf8');
 }
 
-//  Used to compare files expected to be generated by the build process against the expected files
-function compare(root, expectedSiteRelativePath = 'expected', siteRelativePath = '_site') {
+/**
+ * Compares files generated by the build process against the expected files.
+ * Throws an error if any differences are found.
+ * @param {string} root
+ * @param {string} expectedSiteRelativePath
+ * @param {string} siteRelativePath
+ * @param {string} ignoredPaths - Specify any paths to ignore for comparison, but still check for existence.
+ */
+function compare(root, expectedSiteRelativePath = 'expected', siteRelativePath = '_site', ignoredPaths = []) {
   const expectedDirectory = path.join(root, expectedSiteRelativePath);
   const actualDirectory = path.join(root, siteRelativePath);
 
   const expectedPaths = walkSync(expectedDirectory, { directories: false });
-  const actualPaths = walkSync(actualDirectory, { directories: false });
+  let actualPaths = walkSync(actualDirectory, { directories: false });
+
+  // Check for file existence of ignoredPaths and that they are present in actualPaths
+  if (ignoredPaths.length !== 0 && !_.isEqual(_.intersection(ignoredPaths, actualPaths), ignoredPaths)) {
+    throw new Error('Ignored paths are not present in actual paths!');
+  }
 
   let error = false;
-  if (expectedPaths.length !== actualPaths.length) {
-    throw new Error('Unequal number of files');
+  if ((expectedPaths.length + ignoredPaths.length) !== actualPaths.length) {
```

Comment:
> Besides deleting the plantuml images and git ignore them, I realized I will also need to delete them from the expected folder again, before running the functional test. This is because these files which will still be generated after an npm run updatetest will affect testing.

What do you think about filtering the `expectedPaths` first before comparing so that we do not need to delete the generated puml images? From the perspective of the developers, it feels slightly off to have the images generated after `updatetest` and then only deleted after running `npm run test`. 

Would also be nice to keep the images in the expected folder in case the developers want to have a look at them (they don't have to run `updatetest` to regenerate).

Reply:
Hmm, thanks for the feedback. I'm ok with the filtering of expectedPaths. When doing so, I realized that there's a mismatch between CI and our local environment. The CI environment will never contain the generated images, but the local environment will first contain none of the images, and then have the generated images after one `npm run updatetest`. 

I guess it's ok but just want to point that out.

Reply:
An alternative would be to remove the generated png files in the `updatetest` step. This would ensure consistency between CI and local environment but will make it very difficult for devs to manually check the generated images for correctness. What do you think?

Reply:
Yup that works too, I think weighing the cost, we can go with the previous suggestion so that there's a probability of checking the generated files if needed